python/az/aro/azext_aro/custom.py

```diff
@@ -273,13 +273,19 @@ def generate_random_id():
     return random_id
 
 
-def get_route_tables_from_subnets(cli_ctx, subnets):
+def get_route_tables_from_subnets(cli_ctx, subnets, fail):
     network_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_NETWORK)
 
     route_tables = set()
     for sn in subnets:
         sid = parse_resource_id(sn)
 
+        if not all(key in sid.keys() for key in ('resource_group', 'name', 'resource_name')):
```

Comment:
I don't understand what this test is looking for. Is it checking that resource_group, name and resource_name are keys in sid? How about something like this?
```python3
        if 'resource_group' not in sid or 'name' not in sid or 'resource_name' not in sid:
```